src/Graph.elm

```diff
@@ -730,39 +697,28 @@ type AcyclicGraph n e
     = AcyclicGraph (Graph n e) (List NodeId)
 
 
-unsafeGet : String -> NodeId -> Graph n e -> NodeContext n e
-unsafeGet msg id graph =
-    case get id graph of
-        Nothing ->
-            Debug.crash msg
-
-        Just ctx ->
-            ctx
-
-
-checkForBackEdges : List NodeId -> Graph n e -> Result (Edge e) (AcyclicGraph n e)
+checkForBackEdges : List NodeId -> Graph n e -> Result () (AcyclicGraph n e)
 checkForBackEdges ordering graph =
     let
         check id ( backSet, _ ) =
             let
                 backSetWithId =
                     IntDict.insert id () backSet
 
-                error =
-                    "Graph.checkForBackEdges: `ordering` didn't contain `id`"
-
-                ctx =
-                    unsafeGet error id graph
-
-                backEdges =
-                    IntDict.intersect ctx.outgoing backSetWithId
+                backEdge =
```

Comment:
This is how I dealt with the removal of `unsafeGet`.  However, doing so ended up changing the error case of the returned `Result`.  I am confused why I couldn't construct the Edge in the error case below, but the compiler complained that it was a `Err (Edge ())` instead of a `Err (Edge e)`.  This seems connected to the `()` on line 706 above.  Since we don't know what type `e` is, we can't change that, so I had to change the type of the whole function, and sadly, lose some useful functionality.

Reply:
Note that it's not `Err (Edge ())` but `Err ()`. That totally makes half the use cases of this function useless. The error you're seeing is probably related to the fact that you swapped the arguments to `IntDict.intersect`, which is left biased. We don't want to return the meaningless `()` values of the `backSetWithId`, but rather the values of the `ctx.outgoing` `IntDict`.

Reply:
Good catch!  That works fine now.

Reply:
Though if I use the crashLoop then this isn't necessary to change.  Is that your preference?